crates/turborepo-lib/src/daemon/proto/turbod.proto

```diff
@@ -16,6 +16,15 @@ service Turbod {
   //
   // Since 1.11.0
   rpc DiscoverPackages (DiscoverPackagesRequest) returns (DiscoverPackagesResponse);
+
+  // Request the list of packages that the daemon is aware of.
+  // This is a blocking call that will wait for the daemon to finish
+  // discovering the packages, rather than returning 'unavailable'.
+  // It will _still_ report unavailable if a critical process in the
+  // daemon is not running, and the answer will never come.
+  //
+  // Since 1.12.0
+  rpc DiscoverPackagesBlocking (DiscoverPackagesRequest) returns (DiscoverPackagesResponse);
```

Comment:
Pretty sure this won't work with the current setup, the entire grpc server has a 100ms timeout: https://github.com/vercel/turbo/blob/d438018e4e98d2e5f167613fdc56b22dcbe7ba96/crates/turborepo-lib/src/daemon/server.rs#L240

Reply:
Yeah, our handling of timeouts were sub-optimal. I have moved over to a more reasonable 30s timeout default, coupled with a middleware that sets the timeout in the grpc headers for non blocking requests _if the client hasn't specified them_ meaning we also get better handling there too